test/jdk/javax/accessibility/JFileChooserAccessibleDescriptionTest.java

```diff
@@ -0,0 +1,107 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4515031
+ * @key headful
+ * @summary The JFileChooser Dialog itself has no accessible description.
+ * @run main JFileChooserAccessibleDescriptionTest
+ */
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+public class JFileChooserAccessibleDescriptionTest {
+    private static JFrame jFrame;
+    private static JFileChooser jFileChooser;
+    private static JButton jButton;
+
+    private static Robot robot;
+    private static volatile String description;
+    private static volatile int xLocn;
+    private static volatile int yLocn;
+    private static volatile int width;
+    private static volatile int height;
+
+    public static void createGUI() {
+        jFrame = new JFrame("bug4515031 Frame");
+        jFileChooser = new JFileChooser();
+
+        jButton = new JButton("Show FileChooser");
+        jButton.addActionListener(e -> jFileChooser.showDialog(jFrame, null));
+        jFrame.getContentPane().add(jButton);
+        jFrame.setSize(200, 100);
+        jFrame.setLocationRelativeTo(null);
+        jFrame.setVisible(true);
+    }
+
+    public static void doTest() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            robot = new Robot();
+            robot.setAutoDelay(200);
+            robot.setAutoWaitForIdle(true);
+
+            SwingUtilities.invokeAndWait(() -> {
+                xLocn = jButton.getLocationOnScreen().x;
+                yLocn = jButton.getLocationOnScreen().y;
+                width = jButton.getSize().width;
+                height = jButton.getSize().height;
+            });
+
+            robot.mouseMove(xLocn + width / 2, yLocn + height / 2);
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            SwingUtilities.invokeAndWait(() -> description =
+                jFileChooser.getAccessibleContext().getAccessibleDescription());
+
+            robot.keyPress(KeyEvent.VK_ALT);
+            robot.keyPress(KeyEvent.VK_C);
+            robot.keyRelease(KeyEvent.VK_C);
+            robot.keyRelease(KeyEvent.VK_ALT);
```

Comment:
What is the purpose of the pressing Alt-C here? Looks like on Mac OS X it does nothing and when i comment them out the test runs in the same way as with it.

Reply:
As suggested, I have removed this redundant code and verified it on the windows,mac and linux platforms to be working fine.